node/network/availability-distribution/src/lib.rs

```diff
@@ -360,18 +365,28 @@ where
 			.map(|(peer, _view)| peer.clone())
 			.collect();
 
+		// If we have no peers that are interested, skip the rest.
+		if peers.is_empty() {
```

Comment:
I don't think this optimization will work. AFAIK we have no other place where we would load the chunks other than the code below.

`send_tracked_gossip_message_to_peers` should place the messages in the message vault even though `peers.is_empty()`, but now that I look closer, it sadly doesn't.

So this change preserves current behavior, but the current behavior is slightly broken.

Reply:
I see what you mean! Let me write a test and fix this!

Reply:
Just removing this one is not enough, see line `:454`